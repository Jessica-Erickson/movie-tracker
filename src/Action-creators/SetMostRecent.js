export const setMostRecent = (movies) => ({
  type: 'SET_MOST_RECENT',
  movies: movies.results
})