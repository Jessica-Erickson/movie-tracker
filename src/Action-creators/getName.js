export const getName = (name) => ({
  type: 'GET_NAME',
  name
})