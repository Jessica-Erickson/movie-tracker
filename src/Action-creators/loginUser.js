export const loginUser = (userName) => ({
  type: 'LOGIN_USER',
  userName,
})