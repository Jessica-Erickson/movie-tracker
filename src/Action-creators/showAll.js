export const showAll = () => ({
  type: 'SHOW_ALL'
})