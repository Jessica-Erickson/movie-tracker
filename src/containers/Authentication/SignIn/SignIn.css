@import url('https://fonts.googleapis.com/css?family=Roboto+Slab');

* {
  font-family: 'Roboto Slab', serif;
}

.sign-in-form {
  background-color: #CACAD1;
  border-radius: 10px;
  border-color:  #37424E;
  border-width: 10px;
  box-shadow: 10px 9px 20px 0px;
  display: flex;
  flex-direction: column;
  margin-top: 20px;
  margin-left: 20px;
  padding-bottom: 20px;
  padding-top: 10px;
  position: absolute;
  width: 250px;
  z-index: 10;
}

.email {
  border-color:  #37424E;
  border-radius: 5px;
  display: block;
  font-family: 'Roboto Slab', serif;
  height: 1.25rem;
  margin: 10px auto;
  padding-left: 5px;
  width: 75%;
}


.password {
  border-color:  #37424E;
  border-radius: 5px;
  display: block;
  font-family: 'Roboto Slab', serif;
  height: 1.25rem;
  margin: 10px auto;
  padding-left: 5px;
  width: 75%;
}

.submit {
  background-color: #990000;
  border-radius: 5px;
  color: #CACAD1;
  display: block;
  font-size: 1.25rem;
  margin: 10px auto;
  width: 76%;
  height: 2rem;
}

.sign-up {
  color: #990000;
  display: block;
  font-size: 1.25rem;
  margin: 10px auto;
  text-decoration: none;
}

.error {
  color: #990000;
  font-family: 'Roboto Slab', serif;
}