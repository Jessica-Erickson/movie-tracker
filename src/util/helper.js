import { key } from './key.js';

export const getMovieData = async () => {
  try {
    const response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?api_key=${key}&language=en-US&page=1`)
    const data = await response.json()
    return cleanMovieData(data)
  } catch (e) {
    return e
  }
}

const cleanMovieData = (data) => {
  return data.results.map( movie => ({ 
    title: movie.title,
    poster_path: movie.poster_path,
    overview: movie.overview,
    release_date: movie.release_date,
    favorited: false,
    id: movie.id,
    vote_average: movie.vote_average,
  }))
}

const makeOptions = (user) => {
  let options;

  if (user.name) {
    const { email , password , name } = user;
    options = {
      method: 'POST',
      body: JSON.stringify({
        name,
        password,
        email,
      }),
      headers: {
        'Content-Type': 'application/json'
      },
    }
  } else {
    const { email, password } = user;
    options = {
      method: 'POST',
      body: JSON.stringify({
        password,
        email
        }),
      headers: {
        'Content-Type': 'application/json'
      },
    }
  }

  return options;
}

export const getUser = async (user) => {
  const url = 'http://localhost:3000/api/users';
  const options = makeOptions(user);
  const response = await fetch(url, options);
  if(response.ok) {
    const existingUser = await response.json();
    const { name, id } = existingUser.data;
    return { name, id};  
  } else {
    throw new Error('Email and Password do not match.')  
  }
}


export const addUser = async (user) => {
  const url = 'http://localhost:3000/api/users/new';
  const options = makeOptions(user);
  const response = await fetch(url, options);

  if (response.ok) {
    const newUser = await response.json();

    return {
      name: user.name,
      id: newUser.id
    };
  } else {
    throw new Error('Email has already been used.');
  } 
}

export const loginUser = async (user) => {
  let currentUser;
  if (user.name) {
   currentUser = await addUser(user);
  } else {
   currentUser = await getUser(user);
  }
  return currentUser;
}

export const getFavorites = async (id) => {
  const url = `http://localhost:3000/api/users/${id}/favorites`
  const response = await fetch(url);
  if(response.ok) {
    const favoriteMovies = await response.json();
    return favoriteMovies.data;
  }
}
