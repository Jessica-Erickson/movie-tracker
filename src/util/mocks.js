import { key } from './key.js'

export const fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(mock.mockResults)
      }))

export const mockResults = { 
  results: [
    { 
      poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
      title: 'Venom',
      vote_average: 5 ,
      overview: 'This movie is RAD!',
      release_date: '2018-9-11',
    }
  ] 
}

export const mockMovie = { 
      poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
      title: 'Venom',
      vote_average: 5 ,
      overview: 'This movie is RAD!',
      release_date: '2018-9-11',
    }

export const mockState = { 
    movies: [
      { 
        poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
        title: 'Venom',
        vote_average: 5 ,
        overview: 'This movie is RAD!',
        release_date: '2018-9-11',
      }
    ],
    filter: 'favorites' 
  };

export const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${key}&language=en-US&page=1`;

export const userResponse = {
  data: {
    id: 2,
    name: 'Jessica',
    password: 'passw0rd',
    email: 'jerick999@gmail.com',
  },
  message: 'Retrieved ONE User',
  status: 'success',
}

export const user = {
  password: 'passw0rd',
  email: 'jerick999@gmail.com',
}

export const options = {
  method: 'POST',
  body: JSON.stringify({
  password: 'passw0rd',
  email: 'jerick999@gmail.com',
    }),
  headers: {
    'Content-Type': 'application/json'
  },
}

export const newUserResponse = {
  id: 4,
  message: "New user created",
  status: "success",
}

export const newUser = {
  name: 'Tim', 
  password: 'm30w',
  email: 'tag.garrity@gmail.com', 
}

export const newOptions = {
  method: 'POST',
  body: JSON.stringify({
  name: 'Tim',
  password: 'm30w',
  email: 'tag.garrity@gmail.com',
    }),
  headers: {
    'Content-Type': 'application/json'
  },
}