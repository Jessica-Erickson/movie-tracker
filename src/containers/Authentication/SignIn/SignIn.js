import React, { Component }  from 'react';
import * as API from '../../../util/helper';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom'
import { errorReceived } from '../../../Action-creators/errorReceived';
import { getId } from '../../../Action-creators/getId';
import { getName } from '../../../Action-creators/getName';

export class SignIn extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
    }
  }

  updateValue = (e) => {
    const { value, name } = e.target;
    this.setState({
      [name]: value
    })
  }

  loginUser = async (e) => {
    e.preventDefault();
    let currentUser;
    try {
      currentUser = await API.loginUser(this.state);
      this.props.loginUser(currentUser.id, currentUser.name);
      const faves = API.getFavorites(currentUser.id);
        faves.forEach(faves => {
          if(this.props.movies.some( ( ) => {
            
          }))
        })
      this.props.setError('');
      this.props.history.push('/');
      this.setState({ email: '', password: ''});
    } catch (error) {
      this.props.setError(error.message);
    }
  }

  render() {
    const { email, password } = this.state;
    const { error, id, name } = this.props;

    return (
      <form 
        onSubmit={this.loginUser}>
        { error !== '' && <h2> { error } </h2> }
        <input 
          className='email' 
          onChange={this.updateValue}
          name="email" 
          value={email} 
          placeholder="Enter your email"/>
        <input 
          className='password' 
          onChange={this.updateValue}
          name="password" 
          value={password} 
          type="password" 
          placeholder="Enter your password"/>
        <input 
          type='submit'
          value='Sign In' />
        <NavLink to='/signup'>
          Sign up 
        </NavLink>
      </form>
    )
  }
}

export const mapStateToProps = state => {
  return { 
    id: state.id, 
    error: state.error, 
    name: state.name,
  }
}

export const mapDispatchToProps = dispatch => {
  return {
    loginUser: (id, name) => {
      dispatch(getId(id))
      dispatch(getName(name))
    },
    setError: (error) => {
      dispatch(errorReceived(error))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
