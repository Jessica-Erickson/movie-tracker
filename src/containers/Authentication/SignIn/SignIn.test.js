import { SignIn , mapStateToProps , mapDispatchToProps } from './SignIn.js';
import React from 'react';
import * as mock from '../../../util/mocks.js';
import { shallow } from 'enzyme';
import * as API from '../../../util/helper'
import { errorReceived } from '../../../Action-creators/errorReceived';
import { getId } from '../../../Action-creators/getId';
import { getName } from '../../../Action-creators/getName';
import { logOut } from '../../../Action-creators/logOut';
import configureMockStore from 'redux-mock-store';

describe('SignIn', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow( <SignIn/> )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateValue on change', async () => {
    const expected = 'j';
    const spy = jest.spyOn(wrapper.instance(), 
    'updateValue');
    wrapper.instance().forceUpdate();
    wrapper.find('.email').simulate('change', {
      target: {
        value: 'j',
        name: 'email',
      }
    });
    expect(spy).toHaveBeenCalled();
    expect(wrapper.state('email')).toEqual(expected);
  });
  it('should call loginUser on submit', () => {
    const spy = jest.spyOn(wrapper.instance(), 
    'loginUser');
    wrapper.instance().forceUpdate();
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });

    expect(spy).toHaveBeenCalled();
  })
  it('should call loginUser on submit', () => {
    API.loginUser = jest.fn()
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });

    expect(API.loginUser).toHaveBeenCalled();
  })

  describe('mapStateToProps', () => {
    it('should return an object', () => {
      const mockState = { 
        name: 'Jessica', 
        id: 9, 
        movies: [], 
        error: '' };
      const expected = { 
        id: 9 , 
        error: '' , 
        name: 'Jessica' };

      const result = mapStateToProps(mockState);

      expect(result).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with all the correct params for loginUser', () => {
      const mockDispatch = jest.fn();
      const idAction = getId(5);
      const nameAction = getName('Justin'); 
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.loginUser(5, 'Justin');

      expect(mockDispatch).toHaveBeenCalledWith(idAction);
      expect(mockDispatch).toHaveBeenCalledWith(nameAction);
    });

    it('should call dispatch with no params for logoutUser', () => {
      const mockDispatch = jest.fn();
      const logoutAction = logOut();
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.logoutUser();

      expect(mockDispatch).toHaveBeenCalledWith(logoutAction);
    });

    it('should call dispatch with the correct params for setError', () => {
      const mockDispatch = jest.fn();
      const errorAction = errorReceived('ERROR ERROR ERROR');
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.setError('ERROR ERROR ERROR');

      expect(mockDispatch).toHaveBeenCalledWith(errorAction);
    });
  });
});