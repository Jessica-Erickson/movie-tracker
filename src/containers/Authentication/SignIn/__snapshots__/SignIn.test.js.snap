// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="email"
    name="email"
    onChange={[Function]}
    placeholder="Enter your email"
    value=""
  />
  <input
    className="password"
    name="password"
    onChange={[Function]}
    placeholder="Enter your password"
    type="password"
    value=""
  />
</form>
`;
